lib/HAPServiceNode.js

```diff
@@ -199,6 +200,11 @@ module.exports = function(RED) {
         return;
       }
 
+      if (node.filter === true && msg.topic !== node.topic) {
```

Comment:
As mentioned in another PR I also wrote this code, before seeing you already done it 😅The only thing that I added, that you don't have is this simple little proposal:

```suggestion
      const topic = node.topic ? node.topic : node.name;
      if (node.filter === true && msg.topic !== topic) {
```

If the `topic` isn't set on the node, but filter by topic is enabled then it doesn't in its raw format make sense. But if we fallback the topic to the name of the node, then it suddenly has a sane behavior. At least in my opinion 😄 
All the power but with some additional convention over configuration.

Reply:
Yes why not, that looks smart. We should maybe add something to the help text, explaining the behaviour.